modules/core/src/main/java/org/apache/ignite/internal/processors/authentication/IgniteAuthenticationProcessor.java

```diff
@@ -310,19 +301,16 @@ public AuthorizationContext authenticate(String login, String passwd) throws Ign
 
                     authFuts.put(msg.id(), fut);
 
-                    ctx.io().sendToGridTopic(rndNode, GridTopic.TOPIC_AUTH, msg, GridIoPolicy.SYSTEM_POOL);
+                    ctx.io().sendToGridTopic(rndNode, TOPIC_AUTH, msg, GridIoPolicy.SYSTEM_POOL);
                 }
 
                 fut.get();
-
-                if (fut.retry())
-                    continue;
-
-                return new AuthorizationContext(User.create(login));
-            }
+            } while (fut.retry());
         }
         else
-            return new AuthorizationContext(authenticateOnServer(login, passwd));
+            authenticateOnServer(login, passwd);
+
+        return new SecurityContextImpl(authCtx.subjectId(), login);
```

Comment:
We should feel fields of `SecuritySubject` like `SecuritySubjectType`, `InetSocketAddress`, and so on.